### PR TITLE
koa-joi-router: add prefix() to Router interface

### PR DESCRIPTION
**Pull request by martinmicunda**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/koajs/joi-router#prefix
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by martinmicunda** [comment]
@wingsbob or @wingsbob could you review this PR? Thanks

File: types/koa-joi-router/index.d.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ declare namespace createRouter {
         routes: Spec[];
         route(spec: Spec|Spec[]): Router;
         middleware(): Koa.Middleware;
+        prefix(path: string): string;
```
Comment (wingsbob) [inline types/koa-joi-router/index.d.ts]: Doesn't return a string, I'm afraid. Looks like https://github.com/koajs/joi-router/blob/master/joi-router.js#L42-L45 means the relevant code is here: https://github.com/alexmingoia/koa-router/blob/master/lib/router.js#L296-L306

So prefix actually returns the private property `router` 😐

**Review by wingsbob — CHANGES_REQUESTED** [review]
prefix return type needs fixing, though I'd accept `void`, as the returned value is potentially not that helpful IMO

File: types/koa-joi-router/index.d.ts (hunk 1 above)
Comment (martinmicunda) [reply types/koa-joi-router/index.d.ts]: @wingsbob I have replaced `string` with `Router` type. I checked `@type/koa-router` and it returns `Router` as the type for `prefix`, relevant code https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/koa-router/index.d.ts#L221

Comment (wingsbob) [reply types/koa-joi-router/index.d.ts]: @martinmicunda Unfortunately, `Router` in koa-router and `Router` in koa-joi-router are different types.

I think the options here are to import the type from `koa-router` to return, or to use the `void` type, as I don't think there's a great use-case for using the return value.

Comment (martinmicunda) [reply types/koa-joi-router/index.d.ts]: @wingsbob ah okay. I have replaced the return type with `void`.

**Review by wingsbob — APPROVED** [review]
LGTM

**Pull request merged**